src/main/java/com/farao_community/farao/gridcapa/data_bridge/health/FtpHealthIndicator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)marc
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.gridcapa.data_bridge.health;
+
+import org.apache.commons.net.ftp.FTPClient;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.integration.ftp.session.DefaultFtpSessionFactory;
+import org.springframework.integration.ftp.session.FtpSession;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+/**
+ * @author Marc Schwitzguebel {@literal <marc.schwitzguebel at rte-france.com>}
+ */
+@Component
+@ConditionalOnProperty(prefix = "data-bridge.sources.ftp", name = "active", havingValue = "true")
+public class FtpHealthIndicator implements HealthIndicator {
+
+    @Value("${data-bridge.sources.ftp.host}")
+    private String ftpHost;
```

Comment:
Maybe extract the configuration in a specific class that could be shared with the FtpSource class?

Reply:
true but I don't know if it won't impact greatly the existing code :o